src/EnergyPlus/ZoneEquipmentManager.cc

```diff
@@ -1811,6 +1811,18 @@ namespace ZoneEquipmentManager {
                         ZoneSizing(DesDayNum, CtrlZoneNum).DOASSupHumRatSeq(TimeStepIndex) = 0.0;
                         ZoneSizing(DesDayNum, CtrlZoneNum).DOASTotCoolLoadSeq(TimeStepIndex) = 0.0;
                     }
+
+                    if (allocated(CalcZoneSizing(DesDayNum, CtrlZoneNum).DOASSupMassFlowSeq)) {
+                        CalcZoneSizing(DesDayNum, CtrlZoneNum).DOASSupMassFlowSeq(TimeStepIndex) = 0.0;
+                        CalcZoneSizing(DesDayNum, CtrlZoneNum).DOASHeatLoadSeq(TimeStepIndex) = 0.0;
+                        CalcZoneSizing(DesDayNum, CtrlZoneNum).DOASCoolLoadSeq(TimeStepIndex) = 0.0;
+                        CalcZoneSizing(DesDayNum, CtrlZoneNum).DOASHeatAddSeq(TimeStepIndex) = 0.0;
+                        CalcZoneSizing(DesDayNum, CtrlZoneNum).DOASLatAddSeq(TimeStepIndex) = 0.0;
+                        CalcZoneSizing(DesDayNum, CtrlZoneNum).DOASSupTempSeq(TimeStepIndex) = 0.0;
+                        CalcZoneSizing(DesDayNum, CtrlZoneNum).DOASSupHumRatSeq(TimeStepIndex) = 0.0;
+                        CalcZoneSizing(DesDayNum, CtrlZoneNum).DOASTotCoolLoadSeq(TimeStepIndex) = 0.0;
+                    }
+
```

Comment:
Identical block as above except this one is for CalcZoneSizing

Reply:
So much repetition here.  Are ZoneSizing and CalcZoneSizing the same struct?  If so it would be nice to just put a method on that struct called zeroStuff() or something.  

```
// in the header
struct Whatever {
    void zeroStuff();
    ...
}
// in the implementation
Whatever::zeroStuff() {
    for (TimeStepIndex = 1; TimeStepIndex <= NumOfTimeStepInDay; ++TimeStepIndex) {
        this->DOASSupMassFlowSeq(TimeStepIndex) = 0.0;
        DOASHeatLoadSeq(TimeStepIndex) = 0.0;
        ...
    }
}
// then this code would just reduce to:
if (allocated(ZoneSizing(DesDayNum, CtrlZoneNum).DOASSupMassFlowSeq)) {
    for (DesDayNum = 1; DesDayNum <= TotDesDays + TotRunDesPersDays; ++DesDayNum) {
        for (CtrlZoneNum = 1; CtrlZoneNum <= NumOfZones; ++CtrlZoneNum) {
            ZoneSizing(DesDayNum, CtrlZoneNum).zeroStuff();	                    
        }
    }
}
if (allocated(CalcZoneSizing(DesDayNum, CtrlZoneNum).DOASSupMassFlowSeq)) {
    for (DesDayNum = 1; DesDayNum <= TotDesDays + TotRunDesPersDays; ++DesDayNum) {
        for (CtrlZoneNum = 1; CtrlZoneNum <= NumOfZones; ++CtrlZoneNum) {
            CalcZoneSizing(DesDayNum, CtrlZoneNum).zeroStuff();	                    
        }
    }
}
```

Although it would be really nice to know ahead of time whether this function would be called with this array already allocated so we could confidently *not* check the one allocated variable.

Reply:
But as for this current fix, it seems to make sense.

Reply:
Yes, all the zone sizing arrays are the same struct, struct ZoneSizingData. Wish I would have thought of that earlier, it would have been simple to do.